contrib/babelfishpg_tds/src/backend/tds/tdsrpc.c

```diff
@@ -600,6 +600,9 @@ SPExecuteSQL(TDSRequestSP req)
 		}
 		PG_CATCH();
 		{
+			MemoryContextSwitchTo(curr);
```

Comment:
this code needs re-evaluation. We have error in try and error in cache. Why we have code after end try, control will never go there? Also, it seems errdetail_params can throws the error. It would be more maintainable to put TRY/CATCH inside it and return appropriate value. That we we can have simple ereport on caller side and need not handle it at N number of places.

Reply:
good point. I will make the change

Reply:
I took a look, due to timing constraint, I will fix the crash in SPExecuteSQL as that is a regression.

The generic handling of `errdetail_params` will be handled by another Jira (will open one). It requires some more deep dive as some callers call it inside `PG_CATCH` and been there since Day 1.